chia/wallet/puzzles/vault_p2_recovery.clsp

```diff
@@ -47,19 +47,22 @@
     )
   )
 
-  (list
-    (list AGG_SIG_ME BLS_PK (sha256tree recovery_conditions))
-    (list CREATE_COIN
-      (create_recovery_puzzlehash
-        P2_1_OF_N_MOD_HASH
-        FINISH_RECOVERY_MOD_HASH
-        P2_SECP_PUZZLEHASH
-        TIMELOCK
-        recovery_conditions
+  (if BLS_PK
+      (list
+        (list AGG_SIG_ME BLS_PK (sha256tree recovery_conditions))
+        (list CREATE_COIN
+          (create_recovery_puzzlehash
+            P2_1_OF_N_MOD_HASH
+            FINISH_RECOVERY_MOD_HASH
+            P2_SECP_PUZZLEHASH
+            TIMELOCK
+            recovery_conditions
+          )
+          my_amount
+        )
+        (list ASSERT_MY_AMOUNT my_amount)
```

Comment:
Not a fan of this change I don't think, if someone doesn't want this path to work, we can just not put this path as a viable one to execute within the puzzle.